File: eecs168/2021summer/2021.07.14.classes-part-2/Triangle.cpp
//Triangle.cpp

#include "Triangle.h"

double Triangle::getBase()
{
	return(base);
}

double Triangle::getHeight()
{
	return(height);	
}

bool Triangle::setBase(double newBase)
{
	if(newBase > 0)
	{
		base = newBase;
		return(true);
	}
	else
	{
		base = 0;
		return(false);
	}
}

bool Triangle::setHeight(double newHeight)
{
	if(newHeight > 0)
	{
		height = newHeight;
		return(true);
	}
	else
	{
		height = 0;
		return (false);
	}
}

File: eecs168/2025summer/2025.07.16.classes-part-3/triangle.py
#triangle.py

class Triangle:
    def __init__(self, new_base, new_height):
        print('__init__ called')
        #member variables
        self.base = new_base
        self.height = new_height

    #Goal: Define an area method
    def area(self):
        ans = (1/2)*self.base*self.height
        return ans

    def is_larger(self, other):
        if self.area() > other.area():
            return True
        else:
            return False
        
